server/game/gameSystems/DamageSystem.ts

```diff
@@ -23,7 +22,11 @@ export class DamageSystem<TContext extends AbilityContext = AbilityContext> exte
     protected override readonly targetTypeFilter = [WildcardCardType.Unit, CardType.Base];
 
     public eventHandler(event): void {
-        event.card.addDamage(event.damage);
+        if (typeof event.damage === 'number') {
```

Comment:
To prevent any weird fallthrough situations due to bugs, let's make this a switch statement that checks for `case 'number'` and `case 'function'` and in the `default` case has Contract.fail(\`Unexpected type: ${typeof event.damage}\`).

Ditto for HealSystem.